src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AGenerator.java

```diff
@@ -273,7 +272,7 @@ public double generateEnergy(Location l, SlimefunItem sf, Config data) {
 						
 						progress.remove(l);
 						processing.remove(l);
-						return 0;
+						return ChargableBlock.getCharge(l);
```

Comment:
This should check "ChargableBlock.isChargable(l)" before-hand, otherwise we can run into issues.